tests/multi_ptr/multi_ptr_convert_assignment_ops.h

```diff
@@ -0,0 +1,124 @@
+        section_name(
+            "Check &operator=(const multi_ptr<value_type, ASP, IsDecorated>&)")
+            .with("T", type_name)
+            .with("src address_space", src_address_space_name)
+            .with("src decorated", src_is_decorated_name)
+            .with("dst address_space", dst_address_space_name)
+            .with("dst decorated", dst_is_decorated_name)
+            .create()) {
+      bool res = false;
+      {
+        sycl::buffer<bool, 1> res_buf(&res, sycl::range<1>(1));
+        sycl::buffer<T> val_buffer(&value, sycl::range<1>(1));
+        queue.submit([&](sycl::handler &cgh) {
+          auto res_acc =
+              res_buf.template get_access<sycl::access_mode::write>(cgh);
+          auto acc_for_mptr =
+              val_buffer.template get_access<sycl::access_mode::read>(cgh);
+          cgh.single_task([=] {
+            const src_multi_ptr_t mptr_in(acc_for_mptr);
+            dst_multi_ptr_t mptr_out;
+
+            mptr_out = mptr_in;
+
+            // Check that second mptr has the same value as first mptr
+            res_acc[0] = *(mptr_out.get_raw()) == acc_for_mptr[0];
+          });
+        });
+      }
+      CHECK(res);
+    }
+
+    SECTION(
+        section_name(
+            "Check &operator=(multi_ptr<value_type, ASP, IsDecorated>&)")
+            .with("T", type_name)
+            .with("src address_space", src_address_space_name)
+            .with("src decorated", src_is_decorated_name)
+            .with("dst address_space", dst_address_space_name)
+            .with("dst decorated", dst_is_decorated_name)
+            .create()) {
+      bool res = false;
+      {
+        sycl::buffer<bool, 1> res_buf(&res, sycl::range<1>(1));
+        sycl::buffer<T> val_buffer(&value, sycl::range<1>(1));
+        queue.submit([&](sycl::handler &cgh) {
+          auto res_acc =
+              res_buf.template get_access<sycl::access_mode::write>(cgh);
+          auto acc_for_mptr =
+              val_buffer.template get_access<sycl::access_mode::read>(cgh);
+          cgh.single_task([=] {
+            src_multi_ptr_t mptr_in(acc_for_mptr);
+            dst_multi_ptr_t mptr_out;
+
+            mptr_out = mptr_in;
+
+            // Check that second mptr has the same value as first mptr
+            res_acc[0] = *(mptr_out.get_raw()) == acc_for_mptr[0];
+          });
+        });
+      }
+      CHECK(res);
+    }
+    }
+};
+
+template <typename T>
+class check_multi_ptr_convert_assign_for_type {
+ public:
+  void operator()(const std::string &type_name) {
+    const auto address_spaces = multi_ptr_common::get_address_spaces();
+    const auto is_decorated = multi_ptr_common::get_decorated();
+
+    // Run test with address_space and decorated for source multi_ptr type and
+    // with address_space and decorated for destination  multi_ptr type
+    for_all_combinations<run_convert_assignment_operators_tests, T>(
+        address_spaces, is_decorated, address_spaces, is_decorated, type_name);
```

Comment:
According to spec resulting multi_ptr should have Space == address_space::generic_space

Reply:
Now destination `multi_ptr` will be constructed only with `address_space::generic_space`, I've done it [here](https://github.com/KhronosGroup/SYCL-CTS/pull/349/commits/9a1ee752c1a3cbb1f73d0fabd321819802baa3d7#diff-aaaedf36f568a5a0b02e7455e96e48b9f6c1f06093a504cca944e7edb623965dR31-R33)